### DISCUSSION TITLE
Comments on Lists

### DISCUSSION THREAD

**Discussion by marcelvanwaaijen** [koltyakov/gosip#42] [Ideas]
Hi, 

I was wondering if it is possible to work with comments on lists somehow and if not, would it be possible to add that?

Thanks,
Marcel.

**Comment by koltyakov** [comment]
Hi @marcelvanwaaijen,

Are you referencing to modern items comment feature? 

- GET `/_api/web/lists/getByTitle('ListWithComments')/items(1)/comments` // all item's comments
- GET `/_api/web/lists/getByTitle('ListWithComments')/items(1)/comments(1)` // a specific comment
- POST `/_api/web/lists/getByTitle('ListWithComments')/items(1)/comments` // add new comment
- ... etc

This is a possibility to add to the library.

**Reply by koltyakov** [reply]
While not covered in a Fluent API, any method can be consumed with the generic helpers https://github.com/koltyakov/gosip#generic-http-client-helper

Usage example:

```go
sp := api.NewSP(client)
generic := api.NewHTTPClient(client)

list := sp.Web().Lists().GetByTitle("ListWithComments")
itemURL := list.Items().GetByID(1).ToURL()

res, err := generic.Get(fmt.Sprintf("%s/comments", itemURL), nil)
if err != nil {
  log.Fatal(err)
}

data, _ := api.NormalizeODataCollection(res)

fmt.Printf("%s\n", data)
```

**Reply by marcelvanwaaijen** [reply]
Thanks! That is very useful. But for a future version, would it be possible to have a Comments() function in the Fluent API? That would be really awesome!

But for now I am already very happy with this solution.

Thanks,
Marcel.
